--- a/okhttp/src/commonJvmAndroid/kotlin/okhttp3/Response.kt
+++ b/okhttp/src/commonJvmAndroid/kotlin/okhttp3/Response.kt
@@ -194,11 +194,7 @@ class Response internal constructor(
    *     dropped.
    */
   @Throws(IOException::class)
-  fun trailers(): Headers {
-    val source = body.source()
-    if (source.isOpen) {
-      source.skipAll()
-    }
-    return trailersSource.get()
-  }
+  fun trailers(): Headers = trailersSource.get()
+

